app/helpers/maintenance_tasks/tasks_helper.rb

```diff
@@ -101,16 +101,33 @@ def csv_file_download_path(run)
       )
     end
 
-    # Return the appropriate field tag for the parameter, based on its type.
-    # If the parameter has a `validates_inclusion_of` validator, return a dropdown list of options instead.
-    def parameter_field(form_builder, parameter_name)
-      inclusion_validator = form_builder.object.class.validators_on(parameter_name).find do |validator|
+    # Returns the appropriate value for validator's inclusion options for a Proc, lambda, symbol, or other types.
+    #
+    # @param object [Task] The object for which the value needs to be resolved.
+    # @param parameter_name [String] The parameter name.
+    #
+    # @return [Object] value of the resolved inclusion option.
+    def resolve_inclusion_value(object, parameter_name)
+      inclusion_validator = object.class.validators_on(parameter_name).find do |validator|
         validator.kind == :inclusion
       end
+      return unless inclusion_validator
+
+      in_option = inclusion_validator.options[:in]
 
+      return in_option.call(object) if in_option.respond_to?(:call)
```

Comment:
> Currently we have a bug with this, where if you use a Proc, the view will fail, we should focus on fixing this IMO

I believe this PR actually does addresses this bug, it was actually what got me looking into this first, but then I thought let me try to support all the possible options `validates_inclusion_of` offers

> I think it's a bit too dangerous to call any random Proc like this.

To address this concern, the following change will fix the bug by not rendering the dropdown altogether
```suggestion
      return if in_option.respond_to?(:call)
```